packages/electron-builder/src/publish/PublishManager.ts

```diff
@@ -380,7 +380,33 @@ export async function getPublishConfigs(packager: PlatformPackager<any>, targetS
   }
 
   debug(`Explicit publish provider: ${JSON.stringify(publishers, null, 2)}`)
-  return await <Promise<Array<PublishConfiguration>>>BluebirdPromise.map(asArray(publishers), it => getResolvedPublishConfig(packager.info, typeof it === "string" ? {provider: it} : it))
+  return await (<Promise<Array<PublishConfiguration>>>BluebirdPromise.map(asArray(publishers), it => getResolvedPublishConfig(packager.info, typeof it === "string" ? {provider: it} : it)))
+    .then(publishConfigs => expandPublishConfigs(packager, publishConfigs))
+}
+
+function expandPublishConfigs(packager: PlatformPackager<any>, publishConfigs: Array<PublishConfiguration>) {
+  return publishConfigs.map(publishConfig => expandPublishConfig(packager, publishConfig))
+}
+
+function expandPublishConfig(packager: PlatformPackager<any>, publishConfig: PublishConfiguration) {
+  if ((<S3Options>publishConfig).bucket != null) {
```

Comment:
I think we can just expand macro for all keys in the object. It also will allow us to remove explicit expand for generic URL (see usages in this file). Could you please do it?